### PR TITLE
Make multiple-workers tests fail if (Shared)Worker is not defined

### LINKED ISSUE #11222: /workers/semantics/multiple-workers/* are flaky/disabled in chromium webkit
http://bocoup.github.io/wpt-disabled-tests-report/

Investigate what's up with this test:

Path | Products | Results | Bugs
-- | -- | -- | --
/workers/semantics/multiple-workers/003.html | chromium webkit | [ Failure Timeout ] [ Skip ] | https://crbug.com/655458 None

These can detect if `Worker` or `SharedWorker` are not defined, and fail the test.

### PR DESCRIPTION
**Pull request by zcorpan**
Fixes #11222.

### CODE REVIEW

**Comment by jugglinmike** [comment]
Verified that this transforms ugly timeouts into beautiful failures in Chromium. Good going!

I'm wondering what you think of this approach:

    w1.onerror = this.step_func(function(e) { assert_unreached(e.message); });

It's slightly less invasive and also capable of capturing errors stemming from `postMessage`. Then again, it introduces a dependency on another functionality of Workers. If the implementation is failing on part of that spec, maybe it's not such a good idea to rely on another part of the same spec.

I'm happy to merge this as is, just curious to hear your thoughts.

**Comment by zcorpan** [comment]
That could work I think, didn't think of that. Probably more elegant. I can try that.

File: workers/semantics/multiple-workers/006.html [hunk 1]
```diff
@@ -2,6 +2,9 @@
 if (location.hash == '#1') {
   onconnect = function(e) {
     var port = e.ports[0];
+    onerror = e => {
+      port.postMessage(String(e))
+    }
```
Comment (jugglinmike) [inline workers/semantics/multiple-workers/006.html]: If `#1` has an error, won't that trigger `w1.onerror`?

Comment (zcorpan) [reply workers/semantics/multiple-workers/006.html]: Not when it's a SharedWorker, in chromium, I found out. Also the spec has a difference between dedicated and shared workers: https://html.spec.whatwg.org/multipage/workers.html#runtime-script-errors-2

Comment (jugglinmike) [reply workers/semantics/multiple-workers/006.html]: Got it

**Review by jugglinmike — APPROVED** [review]

**Pull request merged**